executor/simpletest/simple_test.go

```
@@ -880,6 +884,96 @@ func TestUser(t *testing.T) {
+	require.Error(t, tk.Session().Auth(&auth.UserIdentity{Username: "u6", Hostname: "localhost"}, encodePassword("password"), nil))
+	checkAuthUser(t, tk, "u6", 1)
+	require.NoError(t, tk.Session().Auth(&auth.UserIdentity{Username: "u6", Hostname: "localhost"}, nil, nil))
+	require.Error(t, tk.Session().Auth(&auth.UserIdentity{Username: "u4", Hostname: "localhost"}, nil, nil))
+	require.Error(t, tk.Session().Auth(&auth.UserIdentity{Username: "u4", Hostname: "localhost"}, nil, nil))
+	require.Error(t, tk.Session().Auth(&auth.UserIdentity{Username: "u4", Hostname: "localhost"}, nil, nil))
+	require.Error(t, tk.Session().Auth(&auth.UserIdentity{Username: "u4", Hostname: "localhost"}, nil, nil))
+}
+
+func encodePassword(password string) []byte {
+	pwd := auth.EncodePassword(password)
+	hpwd, _ := auth.DecodePassword(pwd)
+	return hpwd
+}
+
+func createAndCheck(tk *testkit.TestKit, sql, rsJSON, user string) {
+	tk.MustExec(sql)
+	sql = selectSQL(user)
+	tk.MustQuery(sql).Check(testkit.Rows(rsJSON))
+}
+
+func alterAndCheck(t *testing.T, tk *testkit.TestKit, sql string, user string, failedLoginAttempts, passwordLockTimeDays int64) {
+	tk.MustExec(sql)
+	userAttributesSQL := selectSQL(user)
+	resBuff := bytes.NewBufferString("")
+	for _, row := range tk.MustQuery(userAttributesSQL).Rows() {
+		_, _ = fmt.Fprintf(resBuff, "%s\n", row)
+	}
+	checkUser(t, resBuff.String(), failedLoginAttempts, passwordLockTimeDays)
+}
+
+func checkUser(t *testing.T, rs string, failedLoginAttempts int64, passwordLockTimeDays int64) {
+	var ua []userAttributes
+	if err := json.Unmarshal([]byte(rs), &ua); err == nil {
+		require.True(t, ua[0].PasswordLocking.FailedLoginAttempts == failedLoginAttempts)
+		require.True(t, ua[0].PasswordLocking.PasswordLockTimeDays == passwordLockTimeDays)
+	} else {
+		fmt.Println(err)
```

Please return the error, receive the error in the `TestXXX` function, and use `require.NoError(t, err)` to make sure that no error happens. Like
```go
err = alterAndCheck(xxx)
require.NoError(t, err)
```